app/src/main/java/org/dharmaseed/android/TalkRepository.java

```diff
@@ -31,37 +31,42 @@ public Cursor getTalkAdapterData(
             boolean isDownloaded
     )
     {
-        List<String> columns = new ArrayList<>();
-        columns.add(DBManager.C.Talk.TABLE_NAME    + "." + DBManager.C.Talk.ID);
-        columns.add(DBManager.C.Talk.TABLE_NAME    + "." + DBManager.C.Talk.TITLE);
-        columns.add(DBManager.C.Talk.TABLE_NAME    + "." + DBManager.C.Talk.TEACHER_ID);
-        columns.add(DBManager.C.Teacher.TABLE_NAME + "." + DBManager.C.Teacher.NAME);
-        return getTalks(columns, searchTerms, isStarred, isDownloaded);
+
+        return getTalks(searchTerms, isStarred, isDownloaded, "");
     }
 
     /**
-     * @param columns the columns to select or null if all
      * @param searchTerms the term to filter by or null if none
      * @param isStarred whether the talk is starred
      * @param isDownloaded whether the talk is downloaded
+     * @param extraWhere an extra where clause that can be used to filter on a specific teacher/center id
      * @return every talk in the db filtered by search term, is starred, and is downloaded
      */
     private Cursor getTalks(
-           List<String> columns,
            List<String> searchTerms,
            boolean isStarred,
-           boolean isDownloaded
+           boolean isDownloaded,
+           String extraWhere
     )
     {
+        String[][] columns = {
+            {DBManager.C.Talk.TABLE_NAME, DBManager.C.Talk.TITLE},
+            {DBManager.C.Talk.TABLE_NAME, DBManager.C.Talk.TEACHER_ID},
+            {DBManager.C.Teacher.TABLE_NAME, DBManager.C.Teacher.NAME},
+            {DBManager.C.Center.TABLE_NAME, DBManager.C.Center.NAME},
+            {DBManager.C.Talk.TABLE_NAME, DBManager.C.Talk.DURATION_IN_MINUTES},
+            {DBManager.C.Talk.TABLE_NAME, DBManager.C.Talk.RECORDING_DATE},
+        };
```

Comment:
these columns should be passed in by the calling function

Reply:
:+1: Will do